app/assets/javascripts/discourse/app/controllers/preferences/account.js

```diff
@@ -146,6 +150,19 @@ export default Controller.extend(CanCheckEmails, {
       });
   },
 
+  @action
+  showUserStatusModal(status) {
+    showModal("user-status", {
+      title: "user_status.set_custom_status",
+      modalClass: "user-status",
+      model: {
+        status,
+        saveAction: async (s) => this.set("newStatus", s),
```

Comment:
what is the goal of async here ?

Reply:
Wonder if modal shouldn't just wrap in `Promise.resolve()` but ok fine :+1:

Reply:
Oh, I think that should work, I'll test it later separately, thank you!